upgrade to 0.1.4

ausgerechnet opened the pull request:
- AMs
  - bugfix in MI
  - signed version of log-likelihood
  - add log-ratio
- test data
  - add UCS gold
- naming conventions
  - follow Evert (2004: 36) *or* contingency table
- usability:
  - calculate observed and expected frequencies on the fly for AMs
  - add inplace operations
- extend README

Fixes https://github.com/fau-klue/pandas-association-measures/issues/12

ausgerechnet commented:
@martialblog if you're happy please merge and deploy to PyPI

martialblog commented on tests/test_measures_gold.py:
```diff
@@ -0,0 +1,101 @@
+
+def test_frequencies():
+
+    df = read_csv("tests/ucs-gold.ds.gz", comment='#', index_col=0,
```
I'm not a big fan of binary data in a repository, it'll become larger and larger when this changes. I'd prefer to host this somewhere else and download it, or have a smaller not-binary dataset

ausgerechnet replied:
well the gold-data won't change and I doubt that Stefan will implement any new measures in UCS. but I see your point – I'll take a sample of all rows and check in uncompressed files.

martialblog commented on tests/test_readme.py:
```diff
@@ -0,0 +1,52 @@
+
+
+def test_input():
+    df = read_csv("tests/ucs-gold.ds.gz", comment='#', index_col=0,
```
Again, binary data

martialblog commented on association_measures/frequencies.py:
```diff
@@ -1,54 +1,111 @@
-    :param pandas.DataFrame df: Pandas Dataframe containing O11, f1, f2 and N
-    :return: Tuple of pandas.Series containing observed frequencies for tokens
+    :param pandas.DataFrame df: Pandas Dataframe containing O11 or f, f1, f2 and N
+    :return: tuple of pandas.Series (O11, O12, O21, O22)
```
Add Docstring for new ValueErrors

martialblog commented:
Hi, I left some reviews.

Also fixed the unittest, looks good but there are some Lint errors.

Also also, I don't think the inplace option is a good idea, as it probably will be deprecated: https://github.com/pandas-dev/pandas/issues/16529

Cheerio, 
Markus

ausgerechnet commented:
Hi,

I agree on the inplace operation. I added this because your original implementation had some – but not consistent – inplace operations. For improved usability, we should thus return either the tuple of series as a dataframe (which can thus be joined with df.join(result) *or* the complete new df (with all the old columns). what do you think?

Where can I see addtional comments?

P

ausgerechnet commented on association_measures/measures.py:
```diff
@@ -8,9 +8,10 @@
 
 
-choose = np.vectorize(binomial) # pylint: disable=invalid-name
+choose = np.vectorize(binomial)  # pylint: disable=invalid-name
```
[pep 8 guidelines](https://www.python.org/dev/peps/pep-0008/#inline-comments)

martialblog replied:
:+1: Learn something new every day

ausgerechnet commented on setup.py:
```diff
@@ -47,11 +47,11 @@
     long_description = DESCRIPTION
 
-# Load the package's __version__.py module as a dictionary.
+# Load the package's version.py module as a dictionary.
```
ocd

ausgerechnet commented on tests/test_measures.py:
```diff
@@ -48,8 +48,10 @@ def test_dice(sample_dataframe):
     df = invalid_dataframe
 
-    df['am'] = am.dice(df)
-    assert np.isnan(df['am'][0])
```
no idea what I'm doing wrong here …

ausgerechnet commented on tests/test_measures.py:
```diff
@@ -74,8 +76,10 @@ def test_t_score(sample_dataframe):
 
-    df['am'] = am.t_score(df)
-    assert np.isnan(df['am'][0])
+    try:
```
see above

ausgerechnet commented on tests/test_measures.py:
```diff
@@ -96,12 +100,16 @@ def test_z_score(sample_dataframe):
-    assert np.isnan(df['am'][0])
+    try:
+        df['am'] = am.z_score(df)
+    except ValueError:
```
see above

ausgerechnet commented on tests/test_measures.py:
```diff
@@ -174,4 +186,4 @@ def test_with_random_data(random_dataframe):
     assert 'mutual_information' in df.columns
     assert 'log_likelihood' in df.columns
-    assert 'hypergeometric_likelihood' in df.columns
+    # assert 'hypergeometric_likelihood' in df.columns
```
current implementation is super-slow, original implementation is buggy

ausgerechnet commented:
- I removed inplace operations.
-

ausgerechnet commented:
I think we're good here. Can you have another look and merge it or request further changes?

martialblog commented on association_measures/measures.py:
```diff
@@ -5,149 +5,202 @@
 
 
-choose = np.vectorize(binomial) # pylint: disable=invalid-name
+def choose(n, k):
```
Can we just fix the choose function as described and not have a second implementation?

See https://github.com/fau-klue/pandas-association-measures/issues/12

ausgerechnet replied:
yes, we can!

and I did.

martialblog commented on association_measures/measures.py:
```diff
@@ -5,149 +5,202 @@
 
 
-def log_likelihood(df):
+def log_likelihood(freq, signed=True):
```
I would like these functions to be as polymorphic as possible, can we maybe not have a parameter but just return an extra column with the signed valued?

ausgerechnet replied:
I don't really understand the issue of polymorphism here, but I don't really mind.

I just removed the unsigned version, nobody should use this one anyway …

martialblog commented on tests/test_binomial.py:
```diff
@@ -1,8 +1,10 @@
+from association_measures.measures import choose
 
 
+@pytest.mark.skip
```
Again, See https://github.com/fau-klue/pandas-association-measures/issues/12

ausgerechnet replied:
works like a charm with new type

martialblog commented on tests/test_readme.py:
```diff
@@ -0,0 +1,49 @@
+from pandas import read_csv
+
+
+def test_input():
```
These tests don't assert anything?

ausgerechnet replied:
this is my way of creating the output for the README. it doesn't test anything in addition to all the other tests. you can change it if you like!

martialblog commented on tests/test_measures.py:
```diff
@@ -1,177 +1,241 @@
-
-
-# Check if any warnings of errors are thrown. Might be an unstable test
-@pytest.mark.stability
```
Why was this removed?

ausgerechnet replied:
random data may lead to ValueErrors (if e.g. O11 > N) – this test is thus not possible anymore. The alternative would be to return NAN values for wrong rows and throw a warning instead of an error. Do you prefer this?

martialblog replied:
Let me have a closer look at it

martialblog commented:
Also the Python Lint is still failing.

ausgerechnet commented:
- changed some coding according to the pylint conventions
- excluded "too-many-branches" from pylint
- changed pylint naming conventions of variables so that variable names can follow the generally accepted terminology
